src/charm.py

```diff
@@ -419,6 +439,47 @@ def _on_database_event(self, event) -> None:
 
         self._update_workload_container_config(event)
 
+    def _on_pro_airgapped_server_relation_changed(self, event: RelationChangedEvent):
+        """Handle pro-airgapped-server relation-changed event."""
+        self._update_workload_container_config(event)
+
+    def _on_pro_airgapped_server_relation_departed(self, event: RelationDepartedEvent):
+        """Handle pro-airgapped-server relation-departed event."""
+        self._update_workload_container_config(event)
+
+    def _get_available_pro_airgapped_server_address(self, relation: Relation) -> Union[str, None]:
```

Comment:
nit: this is done as `Optional[str]` in Python to signal it may return `None` if nothing is found.